addons/website_mail/static/src/js/website_mail.js

```diff
@@ -0,0 +1,187 @@
+odoo.define('website_mail.thread', function(require) {
+    'use strict';
+
+    var web_editor_base = require('web_editor.base');
+    var ajax = require('web.ajax');
+    var core = require('web.core');
+    var Widget = require('web.Widget');
+
+    var qweb = core.qweb;
+    var _t = core._t;
+
+    /**
+     * Widget WebsiteMailThread
+     *
+     *  - Fetch message fron controller
+     *  - Display chatter: pager, total message, composer (according to access right)
+     *  - Provider API to filter displayed messages
+     */
+    var WebsiteMailThread = Widget.extend({
+        template: 'website_mail.chatter',
+        events: {
+            "click .o_website_mail_pager_btn": 'on_click_pager'
+        },
+
+        init: function(parent, options){
+            this._super.apply(this, arguments);
+            this.options = _.defaults($(parent).data(), {
+                'display_composer': false,
+                'csrf_token': odoo.csrf_token,
+                'message_count': 0,
+                'pager_step': 10,
+                'pager_scope': 5,
+                'pager_start': 1,
+                'is_user_public': true,
+                'is_user_publisher': false,
+                'domain': [],
+            }, options || {});
+            this.set('messages', []);
+            this.set('message_count', this.options['message_count']);
+            this.set('pager', {});
+            this.set('domain', this.options['domain']);
+            this._current_page = this.options['pager_start'];
+        },
+        willStart: function(){
+            var self = this;
+            // load qweb template and init data
+            var data = this._prepare_fetch_message_params();
+            return $.when(
+                ajax.jsonRpc('/website_mail/init', 'call', data),
+                this._load_templates()
+            ).then(function(result){
+                // bind events
+                self.on("change:messages", self, self.render_messages);
+                self.on("change:message_count", self, function(){
+                    self.render_message_count();
+                    self.set('pager', self._pager(self._current_page));
+                });
+                self.on("change:pager", self, self.render_pager);
+                self.on("change:domain", self, self.on_change_domain);
```

Comment:
Why not set events unconditionally or via the class parameter thing? Do you really need to wait for templates to be loaded to hook these?